pkg/validation/app.go

```diff
@@ -103,6 +108,59 @@ func (v *Validator) validateConfig(ctx context.Context, cr v1alpha1.App) error {
 	return nil
 }
 
+func (v *Validator) validateNamespaceConfig(ctx context.Context, cr v1alpha1.App) error {
+	annotations := key.AppNamespaceAnnotations(cr)
+	labels := key.AppNamespaceLabels(cr)
+
+	if annotations == nil && labels == nil {
+		// no-op
+		return nil
+	}
+
+	var apps []v1alpha1.App
+	{
+		lo := metav1.ListOptions{
+			FieldSelector: fmt.Sprintf("metadata.name!=%s", cr.Name),
+		}
+		appList, err := v.g8sClient.ApplicationV1alpha1().Apps(cr.Namespace).List(ctx, lo)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		apps = appList.Items
+	}
+
+	for _, app := range apps {
+		if key.AppNamespace(cr) != key.AppNamespace(app) {
+			continue
+		}
+
+		targetAnnotations := key.AppNamespaceAnnotations(app)
+		if targetAnnotations != nil && annotations != nil {
+			for k, v := range targetAnnotations {
+				originalValue, ok := annotations[k]
+				if ok && originalValue != v {
+					return microerror.Maskf(validationError, "app %#q is modifying annotation %#q in the target namespace %#q to the different value",
```

Comment:
Current message is OK but I think we could add more detail to help the user. 

I'd like to know which app CR has the same label and maybe also what the current value is.